### PR TITLE
use specific commitID for commit/rollback

### PR DESCRIPTION
**Pull request by phoracek**
**What this PR does / why we need it**:

Limit space for races and bugs.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/helper/client.go [hunk 1]
```diff
@@ -124,12 +124,12 @@ func set(desiredState nmstatev1alpha1.State) (string, error) {
...
-	return nmstatectl([]string{"commit"}, "")
+func commit(commitID string) (string, error) {
+	return nmstatectl([]string{"commit", commitID}, "")
 }
 
-func rollback(cause error) error {
-	_, err := nmstatectl([]string{"rollback"}, "")
+func rollback(commitID string, cause error) error {
+	_, err := nmstatectl([]string{"rollback", commitID}, "")
 	return fmt.Errorf("rollback cause: %v, rollback error: %v", cause, err)
```
Comment (qinqon) [inline pkg/helper/client.go]: Can you add commit id to to error and use errors.Wrapf ?

File: pkg/helper/client.go [hunk 2]
```diff
@@ -251,6 +251,15 @@ func defaultGw() (string, error) {
 	})
 }
 
+// After running `nmstatectl set`, the last line contains the commit/rollback ID in format of:
+// Checkpoint: <ID>
+func readCommitID(setOutput string) string {
```
Comment (qinqon) [inline pkg/helper/client.go]: setOutput feels like an action, let's call it outputFromSet or nmstateOutput.

File: pkg/helper/client.go [hunk 3]
```diff
@@ -251,6 +251,15 @@ func defaultGw() (string, error) {
...
+	setOutputLines := strings.Split(strings.TrimSpace(setOutput), "\n")
+	commitIDLine := strings.TrimSpace(setOutputLines[len(setOutputLines)-1])
+	commitID := strings.Split(commitIDLine, " ")[1]
+	return commitID
```
Comment (qinqon) [inline pkg/helper/client.go]: Can we do all this with a regex  and capturing group ? something like "^CheckPoint: (.*)$"

File: pkg/helper/client.go [hunk 4]
```diff
@@ -261,46 +270,48 @@ func ApplyDesiredState(desiredState nmstatev1alpha1.State) (string, error) {
...
+	commitID := readCommitID(setOutput)
+
 	// Future versions of nmstate/NM will support vlan-filtering meanwhile
 	// we have to enforce it at the desiredState bridges and outbound ports
 	// they will be configured with vlan_filtering 1 and all the vlan id range
 	// set
 	bridgesUpWithPorts, err := getBridgesUp(desiredState)
 	if err != nil {
-		return "", rollback(fmt.Errorf("error retrieving up bridges from desired state"))
+		return "", rollback(commitID, fmt.Errorf("error retrieving up bridges from desired state"))
```
Comment (qinqon) [inline pkg/helper/client.go]: Let's start to rall Wrapf little by little is the defacto stuff fro future 1.13 but we can start to use it at 1.12 with github.com/pkg/errors

File: pkg/helper/client.go (hunk 3 above)
Comment (phoracek) [reply pkg/helper/client.go]: I prefer basic tools if it is as simple as this. Regex is powerful but terrible to read.

File: pkg/helper/client.go (hunk 4 above)
Comment (phoracek) [reply pkg/helper/client.go]: That would deserve a refactoring of the module. I don't want to mess too much with this PR.

We should refactor the helper module completely

**Comment by phoracek** [comment]
/override  pull-kubernetes-nmstate-e2e-ocp

The provider failed to be started.

**Comment by phoracek** [comment]
/hold

Seems like this change broke rollback

**Pull request closed without merging**